package/Runtime/Plugins/iOS/NimbusManager.swift

```diff
@@ -0,0 +1,196 @@
+//  Copyright © 2021 Timehop. All rights reserved.
+//
+
+import Foundation
+import NimbusRenderStaticKit
+import NimbusRenderVideoKit
+import NimbusKit
+
+@objc public class NimbusManager: NSObject {
+    
+    @objc public static let shared = NimbusManager()
+    
+    private let kCallbackTarget = "IOSAdManager"
+    
+    private var nimbusAdManager: NimbusAdManager?
+    private var adController: AdController?
+    
+    private var adView: AdView?
+    
+    @objc public func initializeNimbusSDK(publisher: String,
+                                          apiKey: String,
+                                          enableSDKInTestMode: Bool,
+                                          logLevel: Int,
+                                          appName: String,
+                                          appDomain: String,
+                                          bundleId: String,
+                                          storeUrlString: String,
+                                          showMuteButton: Bool) {
+        Nimbus.shared.initialize(publisher: publisher, apiKey: apiKey)
+        
+        Nimbus.shared.logLevel = NimbusLogLevel(rawValue: logLevel) ?? .off
+        Nimbus.shared.testMode = enableSDKInTestMode
+        // Nimbus.shared.logger = // TODO: add logged
+        
+        
+        let videoRenderer = NimbusVideoAdRenderer()
+        videoRenderer.showMuteButton = showMuteButton
+        Nimbus.shared.renderers = [
+            .forAuctionType(.static): NimbusStaticAdRenderer(),
+            .forAuctionType(.video): videoRenderer
+        ]
+        
+        guard let domainUrl = URL(string: appDomain),
+              let storeUrl = URL(string: storeUrlString) else {
+            print("Error initializing Nimbus SDK. A valid URL is required for app domain and store url. Received appDomain: \(appDomain) | storeUrl: \(storeUrlString)")
+            return
+        }
+        
+        NimbusAdManager.app = NimbusApp(
+            name: appName,
+            domain: domainUrl,
+            bundle: bundleId,
+            storeUrl: storeUrl
+        )
+        NimbusAdManager.user = NimbusUser(age: 20, gender: .male)
+    }
+    
+    @objc public func unityViewController() -> UIViewController? {
+        return UIApplication.shared.windows.first { $0.isKeyWindow }?.rootViewController
+    }
+    
+    @objc public func showBannerAd(position: String, bannerFloor: Float) {
+        guard let viewController = unityViewController() else { return }
+        
+        let adFormat = NimbusAdFormat.banner320x50
+        let adPosition = NimbusPosition.footer
+        
+        let request = NimbusRequest.forBannerAd(position: position,
+                                                format: adFormat,
+                                                adPosition: adPosition)
+        request.impressions[0].bidFloor = bannerFloor
+        
+        self.destroyExistingAd() // TODO: confirm if this is the expected
```

Comment:
Lets make sure this doesn't take control away from the Unity using as they will be destroying ads on the C# side if need be.

Reply:
This will be executed only if trying to show another AD. Is that the expected behavior?

Reply:
I can think of a case where a game might have an existing banner ad being displayed and then tries to load a rewarded ad. I am not sure if the existing banner should be destroyed when the rewarded ad is shown or if it should remain on the screen in the background but this seems like something we should leave up to the publisher implementing the SDK.

Reply:
Cool. I will remove it.